packages/modules/viewer-react/src/services/iModel/ViewCreator3d.ts

```diff
@@ -86,17 +78,68 @@ export class ViewCreator3d {
       );
     }
 
-    const props = await this._createViewStateProps(models, options);
+    const defaultViewId = await this._imodel.views.queryDefaultViewId();
+    const props = await this._createViewStateProps(
+      models,
+      defaultViewId,
+      options
+    );
     const viewState = SpatialViewState.createFromProps(props, this._imodel);
     try {
       await viewState.load();
     } catch {}
 
-    if (options?.standardViewId) {
-      viewState.setStandardRotation(options.standardViewId);
-    }
-    const range = viewState.computeFitRange();
-    viewState.lookAtVolume(range, options?.vpAspect);
+    // configure the view
+    IModelApp.viewManager.onViewOpen.addOnce((viewPort: ScreenViewport) => {
+      // check for a custom configurer and execute
+      if (options?.viewportConfigurer) {
+        options.viewportConfigurer(viewPort);
+        return;
+      }
+
+      // failing that, if there is a valid default view id, adjust the volume but otherwise retain the view as is
+      if (Id64.isValidId64(defaultViewId)) {
+        if (options?.standardViewId) {
+          viewState.setStandardRotation(options.standardViewId);
+        }
+        const range = viewState.computeFitRange();
+        viewState.lookAtVolume(range, options?.vpAspect);
+        return;
+      }
+
+      // no default view and no custom configurer
+      // default execute the fitview tool and use the iso standard view after tile trees are loaded
+      const tileTreesLoaded = () => {
+        return new Promise((resolve, reject) => {
+          const start = new Date();
+          const intvl = setInterval(() => {
+            if (viewPort.areAllTileTreesLoaded) {
+              clearInterval(intvl);
+              resolve(true);
+            }
+            const now = new Date();
+            // after 20 seconds, stop waiting and fit the view
+            if (now.getTime() - start.getTime() > 20000) {
+              reject();
+            }
+          }, 100);
+        });
+      };
+
+      tileTreesLoaded()
+        .then(() => {
```

Comment:
Why not `.finally` if both the `.then` and `.catch` does the same thing?

Reply:
yes definitely

Reply:
updated